src/lib/OpenEXR/ImfStandardAttributes.h

```diff
@@ -193,35 +260,396 @@ IMF_STD_ATTRIBUTE_DEF (altitude, Altitude, float)
+// shift equal to that effected by a Kodak CC035M or Rosco E-Colour
+// Eighth Minus Green filter.
+//
+// As with colorTemperature, the tint attribute's value is primarily
+// forensic.
+//
+// If the tint attribute is present, the colorTemperature attribute
+// is required to give context to the tint.
+//
+
+IMF_STD_ATTRIBUTE_DEF (tint, Tint, float)
+
+
+//
+// lensMake -- manufacturer or vendor of the lens
+//
+// If present, the value should contain only printable characters
+// and have a nonzero length.
+//
+
+IMF_STD_ATTRIBUTE_DEF (lensMake, LensMake, std::string)
+
+
+//
+// lensModel -- model name or model number of the lens
+//
+// If present, the value should contain only printable characters
+// and have a nonzero length.
+//
+
+IMF_STD_ATTRIBUTE_DEF (lensModel, LensModel, std::string)
+
+
+//
+// lensSerialNumber -- serial number of the lens
+//
+// If present, the value should contain only printable characters
+// and have a nonzero length. Note that despite the name, the
+// string can include non-digits as well as digits.
+//
+
+IMF_STD_ATTRIBUTE_DEF (lensSerialNumber, LensSerialNumber, std::string)
+
+
+//
+// lensFirmwareVersion -- firmware version of the lens
+//
+// If present, the value should contain only printable characters
+// and have a nonzero length.
+//
+
+IMF_STD_ATTRIBUTE_DEF (lensFirmwareVersion, LensFirmwareVersion, std::string)
+
+
+//
+// lensTemperature -- Celsius temperature of the lens at time the
+// image was captured
+//
+
+IMF_STD_ATTRIBUTE_DEF (lensTemperature, LensTemperature, float)
+
+
+//
+// lensAttributes -- lens metadata not specified in other predefined
+// attributes (such as aperture, fNumber, focus, focalLength, lensMake,
+// tStop, etc)
+//
+// This attribute should not be used in place of any predefined
+// lens-related attribute. If present, the string should have a nonzero
+// length.
+//
+
+IMF_STD_ATTRIBUTE_DEF (lensAttributes, LensAttributes, std::string)
+
+
+//
+// focalPlaneImageSize -- width and height in millimeters of the area
+// on the focal plane corresponding to the pixels in the data window,
+// with x defining the width and y defining the height
+//
+
+IMF_STD_ATTRIBUTE_DEF (focalPlaneImageSize, FocalPlaneImageSize, IMATH_NAMESPACE::V2f)
+
+
+//
+// focus -- camera's focus distance, in meters
+//
+// Focus is measured from the camera's film or sensor plane. If the 
+// lens was focused at its hyper-focal setting, the value may be the
+// hyper-focal distance or a positive-infinity floating-point value.
 //
 
 IMF_STD_ATTRIBUTE_DEF (focus, Focus, float)
 
 
+//
+// convergenceDistance -- distance in meters from the baseline of the
+// two lens entrance pupils to the point where the lens optical center
+// axes cross each other
+//
+// Positive values indicate a point in front of the baseline of the
+// two entrance pupils. Orthogonal stereo images represent their
+// infinitely distance convergence distance with a value of positive
+// infinity.
```

Comment:
Here is the 2065-4:2013 text:
![image](https://user-images.githubusercontent.com/1134845/103464164-9085ee80-4cff-11eb-8df1-a85cd75e08a8.png)

It is true that there is implied convergence symmetry. I have looked at the relevant diagram in 2065-4:2013 (and its counterpart in our "Technical Introduction to OpenEXR") for so long that the diagram is turning into a Necker cuboid on me, but I think you could accommodate the vertical displacement by either cameraPosition (2065-4-style) or worldToCamera (reference implementation style). Keystoning is harder but I think you could do it with the cameraViewingDirection + cameaUpDirection vectors.

It may be that if you have the two cameras' cameraPosition, cameraViewingDirection and cameraUpDirection vectors, that convergenceDistance could be a derived value. That said, and there were too many color scientists and production executives in the room and an insufficient number of VFX supervisors to say this, we thought that convergence distance was something people were writing down on camera reports even when they hadn't established a coordinate frame for the set, the camera positions and the camera viewing and up directions.

![image](https://user-images.githubusercontent.com/1134845/103464882-2e2fec80-4d05-11eb-92ca-811341d4b959.png)

![image](https://user-images.githubusercontent.com/1134845/103464893-39831800-4d05-11eb-8434-8ff41a424a42.png)

I have had a couple of VFX supervisors (Dave Stump ASC, Erik Nash) look over this language but will go back to them with your comment, if that's OK with you.

Reply:
Yes, that would be welcome! These diagrams are very helpful. I understand the motivation behind the convergence distance now. I can imagine in the heat of the moment, running a tape measure out to get the convergence distance will provide an inexact, but useful, prior for solving the stereo cameras, and it would be much better to record that, than give up because precise measurements aren't achievable while the production clock is running.

Reply:
This diagram had me flummoxed for the longest time. And the accompanying documentation doesn't help when it says of the camera "… at location {0, 0, 0} and shall look along the positive z axis. The way I read this diagram, the Z axis is punching through a plane that's one unit in Z away from the origin at the point of intersection (hence the '1' just above the word 'window'). But if C is the center of the screen window, and forgive my unbelievable rustiness with 3D geometry, then either the screen window stays in a plane perpendicular to the Z axis but is a trapezoidal screen window instead of a rectangle, or the screen window is rectangular and is perpendicular to a vector from the origin through C, in which case, "shall look along the positive Z axis" should be "shall look vaguely along the positive Z axis", said language not being in SMPTE lexicon.

I wouldn't mind your [i.e. the union of people reading this] opinion(s) on this point. Part of me wants to see if I can make a DASH/plotly app with controls for moving C around. (Thinking about this some more, it feels like moving C has to mean diverting from looking straight down the Z axis, because the display window is always rectangular and going from a trapezoidal screen window to a rectangular display window would involve a resample.)

Reply:
My read is that C is just trivially the Normalized Device Coordinate transform from OpenGL, where the transform is Identity in the upper 3x3, and a translation row [Cx, Cy, Sz, 1], where Sz is 1.

Reply:
You're right, of course. I was thinking myself into a corner because I was imagining the camera constrained to be pointing at C at all times. My bad.